### PR TITLE
[TR] PIM-5384: moved hooks into a dedicated class

### PR DESCRIPTION
**Pull request by solivier**
This PR aims to move all behat hooks into a dedicated class.
Behats are ok for CE and EE orm, odm is running.

http://core-ci.akeneo.com/job/dci_behat/219/
http://core-ci.akeneo.com/job/dci_behat/176/console

### CODE REVIEW

**Comment by jjanvier** [comment]
GTM

**Comment by jmleroux** [comment]
gtm

**Comment by jmleroux** [comment]
We could also use traits to provide some generic methods like ` $this->getMainContext()->getSubcontext('hook')` or others subcontexts

**Comment by solivier** [comment]
@jmleroux yep that seems to be a good idea, maybe on next step

File: features/Pim/Behat/Context/HookContext.php [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+use Behat\Behat\Event\BaseScenarioEvent;
+use Behat\Behat\Event\StepEvent;
+use Behat\Mink\Driver\Selenium2Driver;
+use Behat\Mink\Exception\UnsupportedDriverActionException;
+use Context\SelectiveORMPurger;
+use Doctrine\Common\DataFixtures\Purger\MongoDBPurger;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Yaml\Parser;
+
+class HookContext extends PimContext
```
Comment (ClementGautier) [inline features/Pim/Behat/Context/HookContext.php]: Add a class comment.

File: features/Pim/Behat/Context/HookContext.php [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    }
+
+    /**
+     * Execute javascript
+     *
+     * @param string $script
+     *
+     * @return bool Success or failure
+     */
+    public function executeScript($script)
```
Comment (ClementGautier) [inline features/Pim/Behat/Context/HookContext.php]: This one is not a hook, make it private if its for DRY purpose.

File: features/Pim/Behat/Context/HookContext.php [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+    {
+        foreach ($this->getSmartRegistry()->getManagers() as $manager) {
+            $manager->clear();
+        }
+    }
+
+    /**
+     * @return \Doctrine\Common\Persistence\ManagerRegistry
+     */
+    public function getSmartRegistry()
```
Comment (ClementGautier) [inline features/Pim/Behat/Context/HookContext.php]: same, not a hook

File: features/Pim/Behat/Context/HookContext.php [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    {
+        return [];
+    }
+
+    /**
+     * Add an error message
+     *
+     * @param string $message
+     */
+    public function addErrorMessage($message)
```
Comment (ClementGautier) [inline features/Pim/Behat/Context/HookContext.php]: not a hook

File: features/Pim/Behat/Context/HookContext.php (hunk 1 above)
Comment (solivier) [inline features/Pim/Behat/Context/HookContext.php]: there is no class comments in behat classes

**Pull request merged**